src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Tests/ServiceProviderEngineScopeTests.cs

```diff
@@ -17,5 +19,26 @@ public void Dispose_DoesntClearResolvedServices()
 
             Assert.Single(serviceProviderEngineScope.ResolvedServices);
         }
+
+        public class CountAsyncDisposableService : IAsyncDisposable
+        {
+            public int DisposeCount { get; private set; }
+            public async ValueTask DisposeAsync()
+            {
+                await Task.Delay(1);
+                DisposeCount++;
+            }
+        }
+
+        [Fact]
+        public async Task DisposeAsync_IsCalledOnce()
+        {
+            var asyncDisposableService = new CountAsyncDisposableService();
+            var serviceProviderEngineScope = new ServiceProviderEngineScope(null);
```

Comment:
Instead of relying more on internals visible to, can we instead write the unit test using the public API? This is how the bulk of our testing is done in dotnet/runtime.

Reply:
Ok done